homeassistant/components/flexpool/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow for Flexpool integration."""
+import logging
+import re
+
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from ...const import CONF_ADDRESS
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ADDRESS): str,
+        vol.Optional("workers"): bool,
+        vol.Optional("pool"): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    address = data["address"]
+    if not re.findall("^0x[a-fA-F0-9]{40}$", address):
+        raise InvalidAddress
+
+    return {"title": address}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Flexpool."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
```

Comment:
This could be typed

Reply:
Hmm, couldn't find any integration that's using types here. Could you give me a hint?

Reply:
```suggestion
    async def async_step_user(self, user_input: dict[str, Any] | None = None) -> FlowResult:
```